gratipay/utils/__init__.py

```diff
@@ -132,6 +133,46 @@ def get_team(state):
     return team
 
 
+def b64encode_s(s):
+    prefix = b''
+    if not isinstance(s, bytes):
+        s = s.encode('utf8')
+    else:
+        # Check whether the string is binary or already utf8
+        try:
+            s.decode('utf8')
+        except UnicodeError:
+            prefix = b'.'
+    return prefix + b64encode(s, b'-_').replace(b'=', b'~')
+
+
+def b64decode_s(s, **kw):
+    def error():
+        if 'default' in kw:
+            return kw['default']
+        raise Response(400, "invalid base64 input")
+
+    try:
+        s = s.encode('ascii')
+    except UnicodeError:
+        return error()
+
+    udecode = lambda a: a.decode('utf8')
+    if s[:1] == b'.':
+        udecode = lambda a: a
+        s = s[1:]
+    s = s.replace(b'~', b'=')
+    try:
+        return udecode(b64decode(s, '-_'))
+    except Exception:
+        try:
+            # For retrocompatibility
+            return udecode(b64decode(s))
+        except Exception:
+            pass
+        return error()
+
+
```

Comment:
> I think it's just to be [safe](http://stackoverflow.com/questions/941468/c-sharp-mvc-trailing-equal-sign-in-url-doesnt-hit-route/941481#941481).

Safe against bugs in ASP.NET MVC? Aspen has bugs, but not that one. :-P

This is a simple enough conversion, though. I'm fine with it. 

> It's useful for providing a more helpful context specific error message.

Why can't the caller `try`/`except` for `Response`?

> bytes/unicode conversion.
> . prefix

Still wrapping my head around this. I'm trying to satisfy myself that, in addition to making sense, this extra complexity doesn't introduce any security vulnerabilities ...